### PR TITLE
Attempt to deflake `The only membership state included in an initial …

### LINKED ISSUE #1050: `The only membership state included in an initial sync is for all the senders in the timeline` is flaky
https://github.com/matrix-org/synapse/pull/10040/checks?check_run_id=2668894009

### PR DESCRIPTION
**Pull request by DMRobertson**
…sync is for all the senders in the timeline`

Hopefully fixes #1050

### CODE REVIEW

File: tests/31sync/15lazy-members.pl [hunk 1]
```diff
@@ -99,9 +99,21 @@
             )
          }, foreach => [ 1 .. 10 ])
       })->then( sub {
+         # Ensure synapse has propagated this message to Alice's synchotron
```
Comment (anoadragon453) [inline tests/31sync/15lazy-members.pl:102]:
```suggestion
         # Ensure synapse has propagated this message to Alice's sync stream
```

"synchrotron" is a bit of an unofficial name that we mostly just use to describe sync workers on Synapse.

File: tests/31sync/15lazy-members.pl [hunk 2]
```diff
@@ -99,9 +99,21 @@
...
+         await_sync_timeline_contains( $alice, $room_id, check => sub {
+            my ($event) = @_;
+            log_if_fail "event", $event;
+            my $result = $event->{type} eq "m.room.message" &&
+               $event->{sender} eq $charlie->user_id &&
+               $event->{content}->{body} eq "Message 10";
+            $result
```
Comment (anoadragon453) [inline tests/31sync/15lazy-members.pl:109]: Is this the equivalent of `return $result;`, similar to rust?

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Looks sensible, just a couple clarifying comments.

Failing tests in Dendrite are unrelated.

File: tests/31sync/15lazy-members.pl (hunk 2 above)
Comment (anoadragon453) [reply tests/31sync/15lazy-members.pl]: Aha, OK. Thanks for the explanation. Also looks like there is no differentiator through the semicolon as I originally thought.

**Review by anoadragon453 — APPROVED** [review]

File: tests/10apidoc/09synced.pl [hunk 3]
```diff
@@ -393,7 +393,7 @@ sub assert_state_room_members_match {
 
    my @members = grep { $_->{type} eq 'm.room.member' } @{ $events };
    @members == scalar @{ $member_ids }
-      or die "Expected only ".(scalar @{ $member_ids })." membership events";
+      or die "Expected only ".(scalar @{ $member_ids })." membership events (got ".(scalar @members).")";
```
Comment (richvdh) [inline tests/10apidoc/09synced.pl:396]: for future reference, it's generally easier to replace this sort of construction with a call to `assert_eq`. (`assert_*` didn't exist very early in sytest's history, so this sort of `foo == bar or die` construction typically predates its existence)

**Pull request merged**